blueprints/incredible-squaring/src/lib.rs

```diff
@@ -1,17 +1,22 @@
+use gadget_sdk::clients::tangle::runtime::TangleClient;
+use gadget_sdk::event_listener::tangle::{TangleJobEvent, TangleEventListener};
 use gadget_sdk::job;
-use std::convert::Infallible;
+use gadget_sdk::tangle_subxt::tangle_testnet_runtime::api::runtime_types::pallet_services::module::Event;
 
 /// Returns x^2 saturating to [`u64::MAX`] if overflow occurs.
 #[job(
     id = 0,
     params(x),
     result(_),
     event_listener(
-        listener = TangleEventListener,
-        event = JobCalled,
+        listener = TangleEventListener<Event, TangleClient>,
+        event = Event, // Without specifying a pre-processor, it will default to TangleJobEvent
```

Comment:
Still in debugging phase. This comment is here for me mostly. This will be clarified once we are ready for review.

Overall, we want to actually default to identity for preprocessing and no-op for postprocessing.